pkg/asset/manifests/azure/cluster.go

```diff
@@ -93,25 +66,50 @@ func GenerateClusterAssets(installConfig *installconfig.InstallConfig, clusterID
 		nodeSubnetID = "UNKNOWN"
 	}
 
+	//apiServerLBFrontendConfigName := azure.InternalFrontendIPv4ConfigName
+	apiServerLBFrontendConfigName := fmt.Sprintf("%s-internal-frontEnd", clusterID.InfraID)
+	//apiServerBackendPoolName := azure.InternalBackendAddressPoolIPv4Name
+	apiServerBackendPoolName := fmt.Sprintf("%s-internal", clusterID.InfraID)
+	if installConfig.Config.Networking.IsIPv6() && !installConfig.Config.Networking.IsDualStack() {
+		//apiServerLBFrontendConfigName = azure.InternalFrontendIPv6ConfigName
+		apiServerLBFrontendConfigName = fmt.Sprintf("%s-internal-frontEnd-ipv6", clusterID.InfraID)
+		//apiServerBackendPoolName = azure.InternalBackendAddressPoolIPv6Name
+		apiServerBackendPoolName = fmt.Sprintf("%s-internal-ipv6", clusterID.InfraID)
+	}
 	apiServerLB := capz.LoadBalancerSpec{
 		Name: fmt.Sprintf("%s-internal", clusterID.InfraID),
 		BackendPool: capz.BackendPool{
-			Name: fmt.Sprintf("%s-internal", clusterID.InfraID),
+			Name: apiServerBackendPoolName,
 		},
 		LoadBalancerClassSpec: capz.LoadBalancerClassSpec{
 			Type: capz.Internal,
 		},
 	}
 
+	//controlPlaneBackendPoolName := azure.OutboundBackendAddressPoolIPv4Name
+	controlPlaneBackendPoolName := fmt.Sprintf("%s-outbound-lb-outboundBackendPool", clusterID.InfraID)
+	if installConfig.Config.Networking.IsIPv6() && !installConfig.Config.Networking.IsDualStack() {
+		//controlPlaneBackendPoolName = azure.OutboundBackendAddressPoolIPv6Name
+		controlPlaneBackendPoolName = fmt.Sprintf("%s-outbound-lb-outboundBackendPool-ipv6", clusterID.InfraID)
+	}
```

Comment:
Ditto: All pool names can be encapsulated into a single method and we need to change this condition to:
`if installConfig.Config.Networking.IsIPv6() && installConfig.Config.Networking.IsDualStack() {`